sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -990,18 +995,30 @@ object GpuOverrides {
     expr[Coalesce] (
       "Returns the first non-null argument if exists. Otherwise, null",
       (a, conf, p, r) => new ExprMeta[Coalesce](a, conf, p, r) {
+        override def isSupportedType(t: DataType): Boolean =
+          GpuOverrides.isSupportedType(t,
+            allowNull = true)
```

Comment:
This is very common in the expressions.  Worth creating an `ExprMetaHandlesNull` to cut down on the boilerplate?  I'm not stuck on it.

Reply:
Decimal is coming which will not overlap 100% with these.  I thought the main point of GpuOverrides.isSupportedType was to let us do this an item at a time.  Once things calm down and we have tested everything fully, then we would drop isNull as being configurable from isSupportedType.